lib/fprotect/sys_init_fprotect.c

```diff
@@ -6,25 +6,32 @@
 
 #include <zephyr/kernel.h>
 #include <zephyr/init.h>
-#include <pm_config.h>
 #include <fprotect.h>
+#include <zephyr/storage/flash_map.h>
 
 #define PRIORITY_LEVEL 0 /* Locking should be performed as soon as possible */
 
-BUILD_ASSERT(PM_ADDRESS % CONFIG_FPROTECT_BLOCK_SIZE == 0,
+#ifdef CONFIG_PARTITION_MANAGER_ENABLED
+#define CODE_PARTITION app
+#else
+#define CODE_PARTITION slot0_partition
```

Comment:
Isn't DT_CHOSEN(zephyr_code_partition) more generic assignment?

Reply:
Unless we have PM and DTS for partitioning - not.

Reply:
> Isn't DT_CHOSEN(zephyr_code_partition) more generic assignment?

needs some additional stuff - on it's way https://github.com/nrfconnect/sdk-zephyr/pull/1705